activiti-bpmn-model/src/main/java/org/activiti/bpmn/model/Process.java

```diff
@@ -279,45 +279,45 @@ public void setEventListeners(List<EventListener> eventListeners) {
   }
 
   @SuppressWarnings("unchecked")
-  public <FlowElementType extends FlowElement> List<FlowElementType> findFlowElementsOfType(Class<FlowElementType> type, boolean goIntoSubprocesses) {
-    List<FlowElementType> foundFlowElements = new ArrayList<FlowElementType>();
-    for (FlowElement flowElement : this.getFlowElements()) {
-      if (type.isInstance(flowElement)) {
-        foundFlowElements.add((FlowElementType) flowElement);
+  public <BaseElementType extends BaseElement> List<BaseElementType> findBaseElementsOfType(Class<BaseElementType> type, boolean goIntoSubprocesses) {
+    List<BaseElementType> foundBaseElements = new ArrayList<BaseElementType>();
+    for (BaseElement baseElement : this.getFlowElements()) {
+      if (type.isInstance(baseElement)) {
+        foundBaseElements.add((BaseElementType) baseElement);
       }
-      if (flowElement instanceof SubProcess) {
+      if (baseElement instanceof SubProcess) {
         if (goIntoSubprocesses) {
-          foundFlowElements.addAll(findFlowElementsInSubProcessOfType((SubProcess) flowElement, type));
+            foundBaseElements.addAll(findBaseElementsInSubProcessOfType((SubProcess) baseElement, type));
         }
       }
     }
-    return foundFlowElements;
+    return foundBaseElements;
   }
 
-  public <FlowElementType extends FlowElement> List<FlowElementType> findFlowElementsInSubProcessOfType(SubProcess subProcess, Class<FlowElementType> type) {
-    return findFlowElementsInSubProcessOfType(subProcess, type, true);
+  public <BaseElementType extends BaseElement> List<BaseElementType> findBaseElementsInSubProcessOfType(SubProcess subProcess, Class<BaseElementType> type) {
+    return findBaseElementsInSubProcessOfType(subProcess, type, true);
   }
 
   @SuppressWarnings("unchecked")
-  public <FlowElementType extends FlowElement> List<FlowElementType> findFlowElementsInSubProcessOfType(SubProcess subProcess, Class<FlowElementType> type, boolean goIntoSubprocesses) {
+  public <BaseElementType extends BaseElement> List<BaseElementType> findBaseElementsInSubProcessOfType(SubProcess subProcess, Class<BaseElementType> type, boolean goIntoSubprocesses) {
 
-    List<FlowElementType> foundFlowElements = new ArrayList<FlowElementType>();
+    List<BaseElementType> foundBaseElements = new ArrayList<BaseElementType>();
     for (FlowElement flowElement : subProcess.getFlowElements()) {
       if (type.isInstance(flowElement)) {
-        foundFlowElements.add((FlowElementType) flowElement);
+          foundBaseElements.add((BaseElementType) flowElement);
       }
       if (flowElement instanceof SubProcess) {
         if (goIntoSubprocesses) {
-          foundFlowElements.addAll(findFlowElementsInSubProcessOfType((SubProcess) flowElement, type));
+            foundBaseElements.addAll(findBaseElementsInSubProcessOfType((SubProcess) flowElement, type));
         }
       }
     }
-    return foundFlowElements;
+    return foundBaseElements;
   }
```

Comment:
@davidcanonieto What is reason to make a breaking change in public method of Process class Api and cascade all the changes into engine implementation? I am worried it may also cause problems in downstream and external projects that are using Bpmn model classes. 

It may be better to add another method in the Process class to handle the export use case if really needed, but it may not be need at all. Just changing method signature should work without breaking changes, i.e.

```java
public <FlowElementType extends FlowElement> List<FlowElementType> findFlowElementsOfType(Class<FlowElementType> type)
```
```java
public <BaseElementType extends BaseElement> List<BaseElementType> findFlowElementsOfType(Class<BaseElementType> type)

```

Reply:
@igdianov I reverted the method's name. It should not break anything. This change was done in order to keep meaningful method names but i understand that it can be a breaking change. Please re-review it. Thank you